libs/auth-api-lib/src/lib/services/delegations.service.ts

```diff
@@ -323,21 +323,43 @@ export class DelegationsService {
    * @param user
    * @returns
    */
-  async findAllIncoming(user: User): Promise<DelegationDTO[]> {
+  async findAllIncoming(
+    user: User,
+    delegationTypes?: DelegationType[],
+  ): Promise<DelegationDTO[]> {
     const client = await this.getClientDelegationInfo(user)
 
     const delegationPromises = []
 
-    if (!client || client.supportsLegalGuardians) {
+    const hasDelegationTypeFilter =
+      delegationTypes && delegationTypes.length > 0
+
+    if (
+      (!client || client.supportsLegalGuardians) &&
+      (!hasDelegationTypeFilter ||
+        delegationTypes?.includes(DelegationType.LegalGuardian))
```

Comment:
Nit pick, but the optional chaining is unnecessary as you have checked it before when setting `hasDelegationTypeFilter`

Reply:
Fixed. TS was happy with this change, which I thought wasn't the case